src/main/java/io/neonbee/data/DataContext.java

```diff
@@ -96,6 +98,105 @@ public interface DataContext {
+     * @return a reference to this DataContext for chaining
+     */
+    DataContext mergeResponseData(Map<String, Object> data);
+
+    /**
+     * Put some arbitrary response data in the context. This will be available in any data verticle that receive the
+     * context.
+     *
+     * @param key   the key for the data
+     * @param value the data, must be compatible to {@link JsonObject#put(String, Object)} and should be primitive,
+     *              otherwise it can't be serialized via the event bus by default.
+     * @return a reference to this DataContext for chaining
+     */
+    DataContext putResponseDataEntry(String key, Object value);
+
+    /**
+     * Get some response data from the context. The data is available in any data verticle that receive the context.
+     *
+     * @param key the key for the data
+     * @param <T> the type of the parameter to be returned
+     * @return the data, as expected from {@link JsonObject#getValue(String)}
+     */
+    <T> T getResponseDataEntry(String key);
+
+    /**
+     * Remove some response data from the context. The data is available in any data verticle that receive the context.
+     *
+     * @param key the key for the data
+     * @param <T> the type of the parameter to be returned
+     * @return the previous data associated with the key, as expected from {@link JsonObject#remove(String)}
+     */
+    <T> T removeResponseDataEntry(String key);
+
+    /**
+     * Returns a map between {@link DataRequest} and received data map.
+     *
+     * @return a map between {@link DataRequest} and received data map
+     */
+    Map<DataRequest, Map<String, Object>> receivedData();
+
+    /**
+     * Sets the response data map for all {@link DataRequest}s.
+     *
+     * @param map a map between {@link DataRequest} and received response data for the request.
+     * @return a reference to this DataContext for chaining
+     */
+    DataContext setReceivedData(Map<DataRequest, Map<String, Object>> map);
+
+    /**
+     * Retrieves the received data map for one {@link DataRequest}.
+     *
+     * @param dataRequest a data request
+     * @return response data map for this request
+     */
+    Map<String, Object> findReceivedData(DataRequest dataRequest);
+
+    /**
+     * Retrieves the received data map for one verticle identified by the verticle name.
+     *
+     * @param qualifiedName qualified name of a Received verticle
+     * @return first received data for the request verticle name, if available
+     */
+    Optional<Map<String, Object>> findFirstReceivedData(String qualifiedName);
+
+    /**
+     * Retrieves all received data maps for all verticles identified by the name.
+     *
+     * @param qualifiedName qualified name of a verticle
+     * @return all received data maps from verticles with the name
+     */
+    List<Map<String, Object>> findAllReceivedData(String qualifiedName);
```

Comment:
if receivedData() would return da DataMap, we could get rid of all these methods as well and the interfaces of DataMap and DataContext would stay consistent regardless of what we do.

Reply:
See above.